pkg/api/addons.go

```diff
@@ -243,16 +251,19 @@ func (cs *ContainerService) setAddonsConfig(isUpdate bool) {
 
 	defaultAzureNetworkPolicyAddonsConfig := KubernetesAddon{
 		Name:    AzureNetworkPolicyAddonName,
-		Enabled: azureNetworkPolicyAddonEnabled(o),
+		Enabled: to.BoolPtr(o.KubernetesConfig.NetworkPlugin == NetworkPluginAzure && o.KubernetesConfig.NetworkPolicy == NetworkPolicyAzure),
```

Comment:
Carried this additional default gate over from the "last mile" "concatenate addons string" business logic area.